cmd/sqlcmd/main.go

```diff
@@ -111,6 +117,12 @@ func main() {
 	vars := sqlcmd.InitializeVariables(!args.DisableCmdAndWarn)
 	setVars(vars, &args)
 
+	initializeConsole := (args.InputFile == nil && args.Query == "") || args.UserName != ""
+	if initializeConsole {
+		setConsole(console.NewConsole(""))
+		defer line.Close()
+	}

```

Comment:
why does this move out of run()? the tests call run() directly.

Reply:
I'm reworking the changes. Will be making `line` a global variable and it will be set via a function.

Reply:
the global variable should probably be a func type.